### PR TITLE
Modernize llarp directory

### PR DESCRIPTION
**Pull request by michael-loki**
Mostly automated clang-tidy pass (with manual fix up / review)

### CODE REVIEW

File: llarp/dns.cpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ getDNSstring(const char *const buffer, uint32_t *pos)
 }
 
 void
```
Comment (majestrate) [inline llarp/dns.cpp]: this file is deprecated

File: llarp/dns.h [hunk 2]
```diff
@@ -14,7 +14,7 @@ extern "C"
   // fwd declr
   struct dnsc_answer_request;
```
Comment (majestrate) [inline llarp/dns.h]: this file is also deprecated

File: llarp/dns.hpp [hunk 3]
```diff
@@ -111,24 +111,24 @@ std::string
 getDNSstring(const char *const buffer, uint32_t *pos);
 
 void
```
Comment (majestrate) [inline llarp/dns.hpp]: as is this file

File: llarp/dns.cpp (hunk 1 above)
Comment (michael-loki) [reply llarp/dns.cpp]: yeah, will rebase on top of #686 once that is merged

File: llarp/net/ip.hpp [hunk 4]
```diff
@@ -85,7 +86,8 @@ namespace llarp
     /// a network layer packet
     struct NetPacket
     {
-      virtual ~NetPacket(){};
+      virtual ~NetPacket() = default;
+      ;
```
Comment (majestrate) [inline llarp/net/ip.hpp]: trailing `;`

**Comment by majestrate** [comment]
lots of merge conflicts :>

**Comment by michael-loki** [comment]
@majestrate ready for re-review

**Comment by majestrate** [comment]
i'll wait for ci to complete

**Comment by michael-loki** [comment]
fucking MSFT - they have a `#define near`.....

**Comment by majestrate** [comment]
lol

**Comment by michael-loki** [comment]
merging. `make windows` and the lint check failing are different issues.

**Pull request merged**